### DISCUSSION TITLE
Local Feature Configuration Panel

### DISCUSSION THREAD

**Discussion by gjaekel** [igniterealtime/openfire-pade-plugin#342] [Ideas]
What's about to implement a panel that allows the participant to optionally override a bunch of aspects that are provided in general by the server via the `config.js` and `interface_config.js` -- like the initial mute state of audio and video, tiling, full screen, chat and filmstrip panel, audio processing options and so on?

Technical, this might be some panel containing "generated" rows with a description and a thee-state slider (forced disabled, default as by server,forced enabled). If this kind of widget isn't available, two sliders (default, override) (disabled,enabled). The actual set of configuration parameter that is offered to override by the local user should be meta-configurable at the Admin UI and send by the server as a map of the names of the config variables to presented and the description.

The user's overrides may be stored using a certain key -- say `features/overrides` -- in the local store as an map of the config variable names as the key and a boolean as value. This represents the local overrides. An absent key represents the "use server default"

This Panel might be invoked by an icon button at the front page, i.e. before joining a conference. It might be also opened via a button inside a conference. In this case, it should also offer a button to rejoin (i.e. reload) to apply changed settings.

Inside a conference room, before joining the conference, the persisted set of overrides may be evaluated and applied.

Something similar is already implemented as _branding_ inside the Pàdé WebClient.

**Comment by deleolajide** [comment]
It is a great idea 👍 
The Pade web client uses the jitsi-meet url with hash config values to override the global config values in config.js

**Reply by gjaekel** [reply]
> The Pade web client uses the jitsi-meet url with hash config values to override the global config values in config.js

I know. I want to refer to Pàdé's options page where each user may configure aspects. But the actual set of the aspects to be configurable or fixed is determined by the server, nevertheless.

**Comment by gjaekel** [comment]
Maybe a usable implementation of a tri-state switch: https://github.com/jchiam/react-three-state-checkbox

**Reply by deleolajide** [reply]
I would suggest improving this page and generate a  jitsi-meet url  with hash config values from a [custom web url protocol](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/registerProtocolHandler) like 
```
web+meet://anyroomyoulike
```
will be converted into something like
```
https://my-server:7443/ofmeet/anyroomyoulike#config.minHDHeight=540&config.resolution=720&config.startBitrate=720&config.disableAudioLevels=true&interfaceConfig.INITIAL_TOOLBAR_TIMEOUT=20000&interfaceConfig.TOOLBAR_TIMEOUT=4000&interfaceConfig.FILM_STRIP_MAX_HEIGHT=90&interfaceConfig.VERTICAL_FILMSTRIP=true
```
![image](https://user-images.githubusercontent.com/110731/119801994-10db0500-bed6-11eb-890f-8c6276cba69c.png)

**Comment by gptev** [comment]
Super idea!

**Reply by gptev** [reply]
Still no progress, guys?

**Reply by gjaekel** [reply]
I'm busy with other things at the moment. In addition: It's summer and holiday time :smile:
